presto-product-tests/src/main/java/com/facebook/presto/tests/hive/HivePartitioningTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests.hive;
+
+import com.facebook.presto.jdbc.PrestoResultSet;
+import com.facebook.presto.tests.querystats.QueryStatsClient;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.context.ThreadLocalTestContextHolder;
+import com.teradata.tempto.query.QueryResult;
+
+import java.sql.SQLException;
+
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.String.format;
+
+public abstract class HivePartitioningTest
+        extends ProductTest
+{
+    protected long getProcessedLinesCount(String sqlStatement, QueryResult queryResult)
+            throws SQLException
+    {
+        QueryStatsClient queryStatsClient = ThreadLocalTestContextHolder.testContext().getDependency(QueryStatsClient.class);
```

Comment:
Using thread-locals to pass state around doesn't seem like a good idea. This usage is eerily similar to a static per-thread lookup, which is also not great because it suffers from most of the problems dependency injection tries to solve (hidden dependencies on external state, etc).

Ideally, you'd inject the `QueryStatsClient` into each test execution. You might want to consider modeling how state holders/dependencies are injected into tests after the way JMH does it via the `@Scope` annotation (not a pre-requisite for this PR, since it'd be a major change to Tempto)